README.md

```diff
@@ -1,29 +1,56 @@
 [![REUSE status](https://api.reuse.software/badge/github.com/kyma-project/modulectl)](https://api.reuse.software/info/github.com/kyma-project/modulectl)
-# modulectl
+# `modulectl`
 
 ## Overview
-`modulectl` is a command line tool which supports Kyma Module developers. It provides a set of commands and flags you can use to:
-- Create an emtpy scaffold for a new module
+It is a command line tool which supports Module developers for Kyma. It provides a set of commands and flags which can
+be used to:
+- Create an empty scaffold for a new module
 - Build a module and push it to a remote repository
 
-## Prerequisites
-> List the requirements to run the project or example.
+## How to Install
 
-## Installation
+From the GitHub releases page (https://github.com/kyma-project/modulectl/releases), download the binary for your operating system and architecture.
+Then, move the binary to a directory in your PATH.
+Or go inside the directory where the binary is located, and you can run the binary from there.
+Don't forget to make the binary executable by running `chmod +x modulectl`.
 
-> Explain the steps to install your project. If there are multiple installation options, mention the recommended one and include others in a separate document. Create an ordered list for each installation task.
+### Alternative
+You can build the binary from the source code.
+Clone the repository and run `make build` from the root directory of the repository.
+The binary will be created in the `bin` directory.
+> **Note**
 >
-> If it is an example README.md, describe how to build, run locally, and deploy the example. Format the example as code blocks and specify the language, highlighting where possible. Explain how you can validate that the example ran successfully. For example, define the expected output or commands to run which check a successful deployment.
->
-> Add subsections (H3) for better readability.
+> We also provide specific Makefile targets for MacOS (darwin) & Linux operating systems, with the options to compile
+> for x86 or ARM architectures.
+> You can use them to build the binary for your specific operating system and architecture.
 
 ## Usage
+```
+modulectl <command> [flags]
+```
+
+### Available Commands
+- `create` - Creates a module bundled as an OCI artifact. Detailed long explanation can be found here.
+- `scaffold` - Generates necessary files required for module creation.
+- `help` - Provides help about any command.
+- `version` - Print the version of the `modulectl` tool.
+- `completion` - Generate the autocompletion script for the specified shell.
+
+For detailed information about the commands, you can use the `-h` or `--help` flag with the command.
```

Comment:
```suggestion
For detailed information about the commands, use the `-h` or `--help` flag with the command. For example, `modulectl create -h`. See [modulectl](./docs/gen-docs/modulectl.md).
```
We can leave it of course, but as it overlaps with what we have in line 35 about the `help` command maybe we can extend the `help` command description instead. As you prefer.

Reply:
Let's leave it. The `help` above is a command, which prints information `modulectl` itself. And the `--help` & `-h` are flags for each individual subcommand like `create` & `scaffold`.